README.md

```diff
@@ -18,8 +18,122 @@ Capabilities provided by `caikit-nlp`:
 
 ### Getting Started
 
+#### Notebooks
+
 To help you quickly get started with using Caikit, we have prepared a [Jupyter notebook](examples/Caikit_Getting_Started.ipynb) that can be run in Google Colab. Caikit-nlp is a powerful library that leverages prompt tuning and fine-tuning to add NLP domain capabilities to caikit.
 
+#### Installation
+
+To install from git repo:
+
+```bash
+python -m venv .venv
+source .venv/bin/activate
+pip install git+https://github.com/caikit/caikit-nlp
+```
+
+#### Converting Models to caikit format
+
+`caikit_nlp` requires models to be converted to a compatible format, here is an example with the [`flan-t5-small`](https://huggingface.co/google/flan-t5-small) model.
+
+First download the model:
+
+```bash
+# Note: this clone requires git-lfs to be installed and set up for the model
+# to be properly downloaded
+git clone https://huggingface.co/google/flan-t5-small
+```
+
+Then convert it:
+
+```python
+import caikit_nlp
+
+model_path="flan-t5-small"
+model = caikit_nlp.text_generation.TextGeneration.bootstrap(model_path)
+model.save(f"{model_path}-caikit")
+```
+
+#### Serving models
+
+To serve models, the following basic configuration can be used:
+
+```yaml
+# config.yml
+runtime:
+  library: caikit_nlp
+  local_models_dir: ./models
+
+log:
+  formatter: pretty # optional: log formatter is set to json by default
+```
+
+Start the server:
+
+```bash
+env CONFIG_FILES=./config.yml python -m caikit.runtime
+```
+
+The model can now be queried at `localhost:8080` via http or grpc.
+
+For example, using the http server and using curl to send a POST request:
+
+```bash
+curl --json '{
+    "model_id": "flan-t5-small-caikit",
+    "inputs": "At what temperature does liquid Nitrogen boil?"
+}' localhost:8080/api/v1/task/text-generation
+```
+
+We get the following response:
+
+```json
+{
+  "generated_text": "74 degrees F",
+  "generated_tokens": 5,
+  "finish_reason": "MAX_TOKENS",
+  "producer_id": {
+    "name": "Text Generation",
+    "version": "0.1.0"
+  },
+  "input_token_count": 10,
+  "seed": null
+}
+```
+
+All the available API endpoints and protos can be dumped using [`scripts/dump_apis.sh`](/scripts/dump_apis.sh).
+
+#### Docker
+
+`caikit-nlp` can also be deployed as a Docker container. To build the docker image:
```

Comment:
as per my suggestion above regarding entrypoint, we would also need to modify this section a bit.

Reply:
Done